stress/shell/src/main/java/alluxio/stress/cli/fuse/FuseIOBench.java

```diff
@@ -320,32 +375,38 @@ private void runInternal() throws Exception {
       mStartBarrierPassed = true;
 
       if (mParameters.mOperation == FuseIOOperation.LIST_FILE) {
+        for (String workerDirNames: mWorkerDirNames) {
+          for (int dirId = mThreadId; dirId < mParameters.mNumDirs; dirId += mParameters.mThreads) {
+            String dirPath = String.format("%s/%s/dir-%d", mParameters.mLocalPath,
+                workerDirNames, dirId);
+            File dir = new File(dirPath);
+            dir.listFiles();
+          }
+        }
+      } else if (mParameters.mOperation == FuseIOOperation.WRITE
+          || mParameters.mOperation == FuseIOOperation.LOCAL_READ) {
         for (int dirId = mThreadId; dirId < mParameters.mNumDirs; dirId += mParameters.mThreads) {
-          String dirPath = String.format("%s/%d", mParameters.mLocalPath, dirId);
-          File dir = new File(dirPath);
-          dir.listFiles();
+          for (int fileId = 0; fileId < mParameters.mNumFilesPerDir; fileId++) {
+            String filePath = String.format("%s/%s/dir-%d/file-%d", mParameters.mLocalPath,
+                mBaseParameters.mId, dirId, fileId);
+            processFile(filePath, isRead);
+          }
         }
-        return;
-      }
-
-      for (int dirId = mThreadId; dirId < mParameters.mNumDirs; dirId += mParameters.mThreads) {
-        for (int fileId = 0; fileId < mParameters.mNumFilesPerDir; fileId++) {
-          mCurrentOffset = 0;
-          String filePath = String.format("%s/%d/%d", mParameters.mLocalPath, dirId, fileId);
-          while (!Thread.currentThread().isInterrupted()) {
-            if (isRead && CommonUtils.getCurrentMs() > mContext.getEndMs()) {
-              closeInStream();
-              return;
-            }
-            long ioBytes = applyOperation(filePath);
-
-            // Done reading/writing one file
-            if (ioBytes <= 0) {
-              break;
-            }
-            // Start recording after the warmup
-            if (CommonUtils.getCurrentMs() > recordMs) {
-              mThreadCountResult.incrementIOBytes(ioBytes);
+      } else {
+        for (int dirRead = 0; dirRead < mWorkerDirNames.size(); dirRead++) {
+          int indexCurrentDir = (dirRead + mWorkerZeroBasedId) % mWorkerDirNames.size();
+          // Skip itself if the operation is remote read.
+          if (indexCurrentDir == mWorkerZeroBasedId
+              && mParameters.mOperation == FuseIOOperation.REMOTE_READ) {
+            indexCurrentDir++;
+          }
+          String nameCurrentDir = mWorkerDirNames.get(indexCurrentDir);
+          for (int dirId = mThreadId; dirId < mParameters.mNumDirs; dirId += mParameters.mThreads) {
+            for (int fileId = mWorkerZeroBasedId; fileId < mParameters.mNumFilesPerDir;
```

Comment:
for these two for loops, can it swap:
now we use thread id to decide dirId, and worker id to decide fileId,
can we do it like use worker id to decide fileId, and thread id to decide fileId

what's the difference between these two approaches? the reason that i suggest swapping is that it will be more engineer intuitive to go from bigger concepts (worker id, dir id) to smaller concepts together (thread id, file id)

Reply:
Makes sense. This is definitely more intuitive.